sync/sync_head.go

```diff
@@ -38,7 +38,11 @@ func (s *Syncer[H]) Head(ctx context.Context, _ ...header.HeadOption[H]) (H, err
 		return s.Head(ctx)
 	}
 	defer s.getter.Unlock()
-	netHead, err := s.getter.Head(ctx, header.WithTrustedHead[H](sbjHead))
+	// limit time to get a recent header
+	// if can't get it - give what we have
+	reqCtx, cancel := context.WithTimeout(ctx, time.Second*2) // TODO(@vgonkivs): make timeout configurable
+	defer cancel()
```

Comment:
Why timeout is so low? Requesting from Bali to EU nodes takes more time. 🙂

Reply:
>we dont want to make the caller wait for too much and give what we have. We tradeoff absolutely correct network header(as that anyway impossible) for faster operation.

This makes me wonder if an enforced timeout is suitable here. It might be better if the timeout is controlled by the caller of Head(), who has more information about call time constraints.

Reply:
Syncer Head assumes that we are getting head not from the network directly. This case is more of a fallback solution if smth went wrong. setting the timeout here helps to avoid such issues that we have when syncer gets stuck.

Reply:
> It might be better if the timeout is controlled by the caller of Head(), who has more information about call time constraints.

This makes sense, but it won't work in this case. The caller won't be able to fallback to existing subjectiveHead in case context is out.